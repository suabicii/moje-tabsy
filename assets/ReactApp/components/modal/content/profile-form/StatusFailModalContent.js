import React from "react";

function StatusFailModalContent({setIsModalOpen}) {
    return (
        <div className="container text-center" role="dialog" data-testid="status-failed">
            <h2 className="text-danger">Błąd! 🙁</h2>
            <p>Zmiana danych nie powiodła się <i className="fa-regular fa-face-frown-slight"></i></p>
            <small className="d-block text-danger">Błędny format danych lub błąd serwera 😠</small>
            <button className="btn btn-danger mt-2" onClick={() => {
                setIsModalOpen(false);
            }}>OK <i className="fa-solid fa-check"></i>
            </button>
        </div>
    );
}

export default StatusFailModalContent;