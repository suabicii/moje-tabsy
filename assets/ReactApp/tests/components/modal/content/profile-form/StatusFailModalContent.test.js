import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import StatusFailModalContent from "../../../../../components/modal/content/profile-form/StatusFailModalContent";

it('should correctly render StatusFailModalContent', () => {
    const renderer = new ReactShallowRenderer();
    const setIsModalOpen = jest.fn();

    renderer.render(<StatusFailModalContent setIsModalOpen={setIsModalOpen}/>);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});