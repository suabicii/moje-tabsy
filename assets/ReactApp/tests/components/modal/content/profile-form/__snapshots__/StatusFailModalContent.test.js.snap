// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render StatusFailModalContent 1`] = `
<div
  className="container text-center"
  data-testid="status-failed"
  role="dialog"
>
  <h2
    className="text-danger"
  >
    Błąd! 🙁
  </h2>
  <p>
    Zmiana danych nie powiodła się 
    <i
      className="fa-regular fa-face-frown-slight"
    />
  </p>
  <small
    className="d-block text-danger"
  >
    Błędny format danych lub błąd serwera 😠
  </small>
  <button
    className="btn btn-danger mt-2"
    onClick={[Function]}
  >
    OK 
    <i
      className="fa-solid fa-check"
    />
  </button>
</div>
`;
